### PR TITLE
add functionality for bouncing shorturls

### PR DESCRIPTION
**Pull request by abizer**
requires https://github.com/ocf/ocflib/pull/154

### CODE REVIEW

**Comment by dkess** [comment]
Do we ever expect this to be used?

**Comment by abizer** [comment]
yes, this is how the shorturls will get redirected. Apache on death will turn `ocf.io/s/whatever` into `www.ocf.berkeley.edu/api/shorturl/whatever` which will return the `HttpResponseRedirect` to the final destination.

File: ocfweb/api/urls.py [hunk 1]
```diff
@@ -3,10 +3,13 @@
...
 from ocfweb.api import session_tracking
+from ocfweb.api import shorturls
+
 
 urlpatterns = [
     path('hours', hours.get_hours_all, name='hours_all'),
     path('hours/today', hours.get_hours_today, name='hours_today'),
     path('lab/desktops', lab.desktop_usage, name='desktop_usage'),
     path('session/log', session_tracking.log_session, name='log_session'),
+    path('shorturl/<slug:slug>', shorturls.bounce_shorturl, name='bounce_shorturl'),
```
Comment (cg505) [inline ocfweb/api/urls.py]: We don't actually have a slug path converter, do we? Shouldn't this just be `'shorturl/<slug>'`?

Reference: https://docs.djangoproject.com/en/2.1/topics/http/urls/#registering-custom-path-converters

Comment (abizer) [reply ocfweb/api/urls.py]: it's built-in, I think. This seems to work fine when I run it in testing.

File: ocfweb/api/shorturls.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
+from django.http import HttpResponseNotFound
+from django.http import HttpResponseRedirect
+from ocflib.misc.shorturls import get_connection
+from ocflib.misc.shorturls import get_shorturl
+
+
+def bounce_shorturl(request, slug):
+    if slug:
+        with get_connection() as ctx:
+            target = get_shorturl(ctx, slug)
```
Comment (dkess) [inline ocfweb/api/shorturls.py]: Does this cause an exception/internal server error if the slug doesn't exist?

Comment (abizer) [reply ocfweb/api/shorturls.py]: no, the ocflib function returns `None`

**Pull request merged**